src/lib/discovery.ts

```diff
@@ -171,67 +180,68 @@ Discovery.prototype.log = function() {
 Discovery.prototype.discoverAllAPIs = function(discoveryUrl, callback) {
   const self = this;
   const headers = this.options.includePrivate ? {} : {'X-User-Ip': '0.0.0.0'};
-  transporter.request({uri: discoveryUrl, headers}, (err, resp) => {
-    if (err) {
-      return handleError(err, callback);
-    }
-
-    async.parallel(
-        resp.items.map(api => {
-          return (cb) => {
-            self.discoverAPI(api.discoveryRestUrl, (e, newApi) => {
+  transporter.request<DiscoverAPIsResponse>(
+      {url: discoveryUrl, headers}, (err, resp) => {
+        if (err) {
+          return handleError(err, callback);
+        }
+        const items = resp.data.items;
+        async.parallel(
```

Comment:
nit: is it possible to `Promise.all` on an array of promises; dropping the dependency on async? It is fine to tackle this in a follow-on though.

Reply:
Async is used all over this mess.  Upgrading to promises is going to be a challenge in and of itself, so I'd rather leave this for a follow up.

Reply:
ack.